### PR TITLE
feat: [CCM-4773]: Add Regex Filtering in Perspectives

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Summary:

Add `LIKE` filter in Perspective Filters and Perspective Rule Builder

##### Jira Links:

https://harness.atlassian.net/browse/CCM-4773

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger rebase

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
retrigger coverage

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
retrigger coverage

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

**Comment by rajarshee-harness** [comment]
trigger cypress

File: src/modules/75-ce/components/PerspectiveBuilderPreview/PerspectiveBuilderPreview.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const PerspectiveBuilderPreview: React.FC<PerspectiveBuilderPreviewProps> = ({
       filters: [
         getViewFilterForId(perspectiveId, true),
         ...getTimeFilters(getGMTStartDateTime(timeRangeObj.from), getGMTEndDateTime(timeRangeObj.to)),
-        ...getRuleFilters(normalizeViewRules(formValues.viewRules))
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveBuilderPreview/PerspectiveBuilderPreview.tsx:77]: Why this change?

Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveBuilderPreview/PerspectiveBuilderPreview.tsx:77]: The query was returning an error when sending the `LIKE` operator in a rule filter, so I was told to pass `idFilter` instead of `ruleFilter`.

**Review by rishik30 — APPROVED** [review]
